### ISSUE TITLE
sidebar scrolling

### ISSUE THREAD

**Issue by tgirod** [TiddlyWiki/TiddlyWiki5#1688]
When the sidebar is too long, a second scrollbar appears to the right - this behaviour is especially visible when displaying the taglist. I find it very inconvenient, as you get two scrollbars (sidebar + window) side by side, which can be a nightmare to navigate.

Now, here is an ugly corner case of this feature (at first I thought it was a display bug):
1. go to empty.html
2. in the sidebar, switch to more > tags
3. create a new tiddler with 5 tags
4. in the sidebar, click on the last tag to display the associated list of tiddlers

Now, you get the sidebar scrolling behaviour, but the bottom of the scrollbar is aligned with the "shadows" vertical tab, in the middle of the page.

**Comment by pmario** [comment]
I did a test with you description. But imo it depends on the screen resolution and the browser zoom setting. So with 1080p and zoom level 1 nothing happens. 

I know of the 2 scrollbars and it is ugly but IMO there isn't an easy way to make it better. We could all switch to safari as a browser, because it renders the scrollbars differently. .. or not at all.

**Comment by tgirod** [comment]
True, what I described is probably browser/zoom dependant.

To reproduce what I saw, you just have to create enough tags on a tiddler so that clicking on the tag will make the popup expand below the bottom of the sidebar (here, the "shadow" tab is the last block) ([screenshot](http://i.imgur.com/N3Hjb4k.png)). Also, reducing the window, I see the popup is too wide, causing horizontal scrolling.

Strangely, if I add several untagged tiddlers to observe the same behaviour when clicking on "untagged", the popup is displayed just fine ([screenshot](http://i.imgur.com/JqPciBY.png)).

**Comment by pmario** [comment]
I'm not sure, which OS you use, but since there is an Arch tab open in your screenshot, I think you are using Arch Linux. ... So the problem seems to be OS FF implementation specific. 

The scroll bars in your screen shot imo are are always ugly, even if there is only one of them. This definitely depends on the OS graphic library. .. So there may be the possibility to tweak them on the OS level. ... which imo isn't possible for win :/

I found some hacks: https://support.mozilla.org/en-US/questions/957337  which would look like this, if we apply one more hack to the sidebar CSS 

![scrollbar-hack](https://cloud.githubusercontent.com/assets/374655/7370751/a014eeda-edbc-11e4-9cc3-bb4c75b4ee0a.gif)

**Comment by pmario** [comment]
The scrollbars both are top right and almost invisible but both usable with a mouse only :/

**Comment by tgirod** [comment]
Just to make sure we understand each other, I'm not arguing about the overall ugliness of FF scrollbars - I'm justing point out a situation where they appear, even though they should not.

So you could not reproduce the problem of the tag popup causing the scrollbar to appear, except for "untagged" ? If it is the case, that's weird. You are right, I'm running archlinux + FF37. I also tried with chromium, same behaviour.

**Comment by Jermolene** [comment]
I agree that the current scrollbar arrangements are unusable in some situations.

I'm interested in refactoring the page display using flexbox so that we can support multiple flexible toolbars attached to any edge of the browser window. One of the side-effects would be to bring the various scrollbars together with their content.

**Comment by pmario** [comment]
... but I don't want to have a scrollbar in the middle of the screen. IMO this would be the worst case of all. Two sidebars on the right are not good, but you get used to it. In the middle of the screen it looks just terrible.

**Comment by tgirod** [comment]
To get back on track, what I was [pointing out originally](http://i.imgur.com/N3Hjb4k.png) is a case where the tag popups are not rendered properly, messing up the sidebar with unecessary scrollbars.

Am I the only one to observe this behaviour ?

**Comment by pmario** [comment]
>Am I the only one to observe this behaviour ?

Seems so. I have never seen this behavioiur. win8.1 latest FF

**Comment by pmario** [comment]
I can reproduce it now. :/ strange

**Comment by pmario** [comment]
The tag list pills are elements within a div .tc-tab-content. 
Clicking a pill opens a dropdown. 
The dropdown is a child with a class .tc-drop-down

like so:

```
<div class="tc-tab-content tc-vertical">
...
  <div class="tc-drop-down">
  </div>
</div>
```

`.tc-tab-content` CSS contains an `overflow:auto` so if the child dropdown exceeds the height and/or the width of the parent. Sliders are drawn. .. That's the reason. 

So with a closer look, there's nothing strange. The browser just does what it was told ;)

If .tc-tab-content.tc-vertical would use `overflow:visible` there would be no scroll bars. ... But since the .tc-tab-content construction is used several times, we would need to check, if there are other side effects if we change it.

**Comment by pmario** [comment]
overflow:auto was introduced with: https://github.com/Jermolene/TiddlyWiki5/commit/42debd489df34d2ddc3fa93517c039d01b8ac0f9
which fixes: "vertical tabs cause overflow problem #1632"

**Comment by Jermolene** [comment]
@pmario it seems that https://github.com/Jermolene/TiddlyWiki5/commit/42debd489df34d2ddc3fa93517c039d01b8ac0f9 does have some unexpected side-effects - which is not surprising, given that the effect of "overflow" is more than just causing the embedded elements to rescale to 100%. Is there really no other way of fixing #1632?

**Comment by pmario** [comment]
>@pmario it seems that 42debd4 does have some unexpected side-effects - which is not surprising, given that the effect of "overflow" is more than just causing the embedded elements to rescale to 100%. Is there really no other way of fixing #1632?

Automatically resizing and preserving the aspect ratio of an embedded video seems to be a general problem. Not only in the TW context. ... I did some search already and found a CSS based solution, that uses a special wrapper. So imo it would be possible to create a macro, that does the magic. ... for embedded videos. 

But I think overflow auto is not the root problem here. ... IMO the root cause of the edge case is, that the dropdown is a reveal widget, that is rendered as part of the tag list DOM. .. Just some magic CSS makes it look like a dropdown. IMO this mechanism will cause more problems in the future too.

**Comment by Jermolene** [comment]
> But I think overflow auto is not the root problem here.

But it introduced a problem that has affected users of TW. There's no easy workaround for them.

> IMO the root cause of the edge case is, that the dropdown is a reveal widget, that is rendered as part of the tag list DOM. .. Just some magic CSS makes it look like a dropdown. IMO this mechanism will cause more problems in the future too.

What else would you propose? In the absence of another solution to implementing dropdowns, I'm inclined to think we should retract https://github.com/Jermolene/TiddlyWiki5/commit/42debd489df34d2ddc3fa93517c039d01b8ac0f9

**Comment by pmario** [comment]
I think #1699 "Proposal: Responsive Video IFrame, that keeps the aspect ratio " can be a solution.

**Comment by Jermolene** [comment]
Hi @pmario but #1699 only fixes #1632 for videos, doesn't it? There must be a cleaner way to make "max-width: 100%" work correctly.

**Comment by pmario** [comment]
FF 64 implements new CSS settings. see  https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-width  

I can't test it atm, since Ubuntu modifies scroll bars on OS level
